zkvm/src/ops.rs

```diff
@@ -184,8 +184,8 @@ impl Instruction {
             Opcode::Mul => Ok(Instruction::Mul),
             Opcode::Eq => Ok(Instruction::Eq),
             Opcode::Range => {
-                let bit_width = program.read_u8()?;
-                Ok(Instruction::Range(bit_width))
+                let bit_width = BitRange::new(program.read_u8()? as usize);
```

Comment:
generally it's a better style to add `.ok_or` before the variable is created, so all variables in the program are maximally "valid".